shared/javascripts/context_messenger.js

```diff
@@ -471,12 +471,20 @@ if (Privly === undefined) {
   SafariAdapter.prototype.setListener = function (callback) {
     if (this.getContextName() === 'BACKGROUND_SCRIPT') {
       safari.application.addEventListener("message", function(payload) {
-        callback(payload);
+        if (typeof payload.name !== "undefined" && payload.name === "privlyMessage") {
+          callback(payload.message);
+        } else {
+          callback(payload);
+        }
       }, true);
     }
     if (this.getContextName() === 'CONTENT_SCRIPT') {
       safari.self.addEventListener("message", function(payload) {
-        callback(payload);
+        if (typeof payload.name !== "undefined" && payload.name === "privlyMessage") {
+          callback(payload.message);
+        } else {
+          callback(payload);
```

Comment:
Added the comments now at https://github.com/privly/privly-applications/commit/f651eb59418ae8c2298db249296b9da804d7b286#diff-e3c7938b324cb640951cd58dd4d34be2R475